python/cudf/dataframe/index.py

```diff
@@ -597,6 +609,10 @@ def __init__(self, values, name=None):
                 categories=values.categories.tolist(),
                 ordered=values.ordered
             )
+        elif isinstance(values, (list, tuple)):
+            values = columnops.as_column(
+                pd.Categorical(values, categories=values)
+            )
```

Comment:
This assumes that `values` is a unique list of elements, is that always the case?

Reply:
That definitely should be the case and the appropriate error message is thrown if it's not:

```
In [3]: a = cudf.dataframe.index.CategoricalIndex(['a', 'b', 'a'])                                                                                                                      
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-1a84850cfd45> in <module>
----> 1 a = cudf.dataframe.index.CategoricalIndex(['a', 'b', 'a'])

~/workspace/cudf/python/cudf/dataframe/index.py in __init__(self, values, name)
    612         elif isinstance(values, (list, tuple)):
    613             values = columnops.as_column(
--> 614                 pd.Categorical(values, categories=values)
    615             )
    616 

~/miniconda3/envs/cudf_dev/lib/python3.7/site-packages/pandas/core/arrays/categorical.py in __init__(self, values, categories, ordered, dtype, fastpath)
    305         else:
    306             # If dtype=None and values is not categorical, create a new dtype
--> 307             dtype = CategoricalDtype(categories, ordered)
    308 
    309         # At this point, dtype is always a CategoricalDtype

~/miniconda3/envs/cudf_dev/lib/python3.7/site-packages/pandas/core/dtypes/dtypes.py in __init__(self, categories, ordered)
    136 
    137     def __init__(self, categories=None, ordered=None):
--> 138         self._finalize(categories, ordered, fastpath=False)
    139 
    140     @classmethod

~/miniconda3/envs/cudf_dev/lib/python3.7/site-packages/pandas/core/dtypes/dtypes.py in _finalize(self, categories, ordered, fastpath)
    161         if categories is not None:
    162             categories = self.validate_categories(categories,
--> 163                                                   fastpath=fastpath)
    164 
    165         self._categories = categories

~/miniconda3/envs/cudf_dev/lib/python3.7/site-packages/pandas/core/dtypes/dtypes.py in validate_categories(categories, fastpath)
    322 
    323             if not categories.is_unique:
--> 324                 raise ValueError('Categorical categories must be unique')
    325 
    326         if isinstance(categories, ABCCategoricalIndex):

ValueError: Categorical categories must be unique
```